app/code/Magento/Catalog/Model/Indexer/Product/Flat/Action/Eraser.php

```diff
@@ -151,4 +174,18 @@ public function deleteProductsFromStore($productId, $storeId = null)
             );
         }
     }
+
+    /**
+     * Get MetadataPool instance
+     *
+     * @return \Magento\Framework\EntityManager\MetadataPool
+     */
+    private function getMetadataPool()
+    {
+        if (null === $this->metadataPool) {
+            $this->metadataPool = \Magento\Framework\App\ObjectManager::getInstance()
+                ->get(\Magento\Framework\EntityManager\MetadataPool::class);
```

Comment:
Please add new dependencies in the constructor method of the class.
The new constructor parameters should be optional in order to avoid backward compatibility issues.
Please check the [Contribution Guide](https://devdocs.magento.com/guides/v2.3/contributor-guide/backward-compatible-development/#adding-a-constructor-parameter) (section *Adding a constructor parameter*) for more details.

Reply:
@dmytro-ch As this was a copy+paste of Magento\Catalog\Model\Indexer\Product\Flat\Action\Indexer do I need to fix it in that one as well?